### ISSUE TITLE
Set the test rule by default to the first rule (root rule) of the parser grammar

### ISSUE THREAD

**Issue by NickStrupat** [antlr/intellij-plugin-v4#460]

**Comment by parrt** [comment]
this makes a lot of sense actually :) thanks for passing along

**Comment by bjansen** [comment]
@parrt have you started working on this one?

Maybe we should target the first toplevel rule (i.e. not referenced by any other rule), instead of simply the first rule in the list?

**Comment by NickStrupat** [comment]
@bjansen Great idea. That's what I was getting at but didn't quite think deeply enough about it.

**Comment by parrt** [comment]
I have not started on that one no.  Yes, I guess it should be the first top-level parsing rule yes; i.e., a rule that is not called by anybody else.  I fixed some important bugs but I will try to focus on 4.10 antlr at the moment if you are able to spend some time on the plug-in.

**Comment by KvanTTT** [comment]
Also such rule should not be marked as `unused parser rule`.

**Comment by parrt** [comment]
OK I'm now starting to work on this.

**Comment by parrt** [comment]
I can't find an easy way in the antlr source code to find references to a rule. Can anybody refresh my memory? It might actually be more intuitive if I simply pick the first rule no matter what in the grammar

**Comment by NickStrupat** [comment]
I believe the first rule is the root in ANTLR but I am not positive.

**Comment by parrt** [comment]
unfortunately it's a bit more complicated. There can be many rules that have no invoking rule. The order of the rules is also not indicative of whether they get referenced. No big deal. We can start with picking the first rule.

**Comment by parrt** [comment]
Anybody care to take a peek? https://github.com/antlr/intellij-plugin-v4/pull/545

**Comment by KvanTTT** [comment]
> There can be many rules that have no invoking rule. The order of the rules is also not indicative of whether they get referenced. No big deal. We can start with picking the first rule.

Maybe the first rule without references (`unused parse rule`)? Or just the first rule in rare cases if there is mutual recursion for all rules.

**Comment by parrt** [comment]
yeah, tough call, but I think I prefer just picking the first rule as it's the most intuitive. Made some updates so that the start rule label gets updated properly.
